Migrate compose WCModalBottomSheetLayout usages to WCModalBottomSheet

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #13495
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
⚠️ Don't panic about the many changes in this PR, but of them are easy to follow. The manual testing will be a bit more cumbersome but it would've been the same if I had splitted the migration done in this PR in several smaller ones. 

We currently have an issue in Android version < 15, where opening any screen that contains/uses `WCModalBottomSheetLayout` will trigger an issues with the status bar scrim color. The issue will start happening after navigating back from the screen containing `WCModalBottomSheetLayout`. 
More details: peaMlT-18m-p2

The bug is caused by [the workaround](https://github.com/woocommerce/woocommerce-android/blob/cb07541879b2484e5f4de90d91a80ecc12eb8588/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/WCModalBottomSheetLayout.kt#L83)we are applying to override the scrim color and ensure that when showing a `ModalBottomSheetLayout`, the "dark overlay" effect is applied edge-to-edge, including the status bar. It seems like this is causing an unwanted side effect: The scrim color of the status bar starts acting weirdly. 

TBH, I couldn't find the root cause for such weird behavior. However, I did find a solution, which also simplifies the implementation of bottom sheets in compose and allows us to get rid of hacky workarounds. The solution migrates to Material 3 [ModalBottomSheet](https://developer.android.com/develop/ui/compose/components/bottom-sheets). This component, which I think was unavailable back when we started adding bottom sheets in compose, works out of the box and has a simpler implementation. 

This PR migrates all usages of `WCModalBottomSheetLayout` in favor of new component `WCModalBottomSheet`, except for 2 of them that will be handled in other [GH issues](https://github.com/woocommerce/woocommerce-android/pull/13506#issuecomment-2646512752): 
- [AddressSelection](https://github.com/woocommerce/woocommerce-android/blob/cb07541879b2484e5f4de90d91a80ecc12eb8588/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/AddressSection.kt#L355)
- [ScanToUpdateInventoryScreen](https://github.com/woocommerce/woocommerce-android/blob/cb07541879b2484e5f4de90d91a80ecc12eb8588/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryScreen.kt#L41)

### Testing information
Review that the following bottom sheets works as expected in light and dark mode: 

| Before  | After |
| ------------- | ------------- |
| <img width="300"  src="https://github.com/user-attachments/assets/b38e228d-f515-4ad7-96b7-7efc7cd3d006">   | <img width="300"  src="https://github.com/user-attachments/assets/76a78192-4b3d-4f05-9497-9cb28ac10a8b"> |
| <img width="300"  src="https://github.com/user-attachments/assets/134c2ef6-c029-49ed-81e7-f0189310e01e">  | <img width="300"  src="https://github.com/user-attachments/assets/5bf23e89-7662-4987-b4fd-1b28a64083fb">  |
| <img width="300"  src="https://github.com/user-attachments/assets/a80c37c6-148e-491e-9e2a-e72890d25fb4">  | <img width="300"  src="https://github.com/user-attachments/assets/12abd45a-4adb-4674-86b7-06711fa91d15">  |
| <img width="300"  src="https://github.com/user-attachments/assets/9848bf87-ee47-40a6-8a36-13e6e059601c">  | <img width="300"  src="https://github.com/user-attachments/assets/ccbb543e-f805-4c24-9bba-2f247f2924f1">  |
| <img width="300"  src="https://github.com/user-attachments/assets/e289289f-7f41-411e-8c46-bf565d6540bf">  | <img width="300"  src="https://github.com/user-attachments/assets/c09cf69e-7ccd-4616-9194-eac8d3f087ac">  |

Screen recording on how to reach some of the bottom sheets (when it's not obvious)

**Change site theme (the site must be WP.com site**
https://github.com/user-attachments/assets/13d32182-7d36-494a-9dc4-5f008afd70ae

**Blaze url add parameter**
https://github.com/user-attachments/assets/842dd65c-f3e1-4e78-9058-55119d092451

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

JorgeMucientes committed:
Add material 3 compose dependency to use ModalBottomSheet

JorgeMucientes committed:
Migrate usage tof Blaze intro screen to use ModalBottomSheet

JorgeMucientes committed:
Adjust Modal bottom sheet background to work well in dark mode

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Add a Woo styled ModalBottomSheet composable component

JorgeMucientes committed:
Arrange WCModalBottomSheet correctly

JorgeMucientes committed:
Use surface elevated as default background for bottom sheets

JorgeMucientes committed:
Replace usage of WCModalBottomSheetLayout for WCModalBottomSheet

Added a task to remove the bottomsheet from this screen because its unsued GH issue #13507

JorgeMucientes committed:
Replace usage of WCModalBottomSheetLayout in Blaze budget screen

JorgeMucientes committed:
Add utility method to dismiss bottomsheet

JorgeMucientes committed:
Use utility method to dismiss bottom sheet

JorgeMucientes committed:
Fix detekt issues

JorgeMucientes committed:
Replace WCModalBottomSheetLayout usage from Blaze destination parameters screen

JorgeMucientes committed:
Remove experimental material annotation in CampaignBudgetScreen.

JorgeMucientes committed:
Refactor ThemePreviewScreen to use WCModalBottomSheet

JorgeMucientes commented:
Hey @samiuelson @atorresveiga 👋🏼 

In this PR I'm working on migrating all the composable bottom sheet usages of `WCModalBottomSheetLayout` a new much simpler component `WCModalBottomSheet` (added in this PR) to fix some UI glitches the app is experiencing. So far I've migrated all the usages of `WCModalBottomSheetLayout` except for 2. Could you help me with those? I'm not even sure how to trigger those bottomsheet from the app: The 2 bottom sheet pending to migrate are in: 
- [AddressSelection](https://github.com/woocommerce/woocommerce-android/blob/cb07541879b2484e5f4de90d91a80ecc12eb8588/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/address/AddressSection.kt#L355) (somewhere in shipping labels flow but could not find how to trigger this)
- [ScanToUpdateInventoryScreen](https://github.com/woocommerce/woocommerce-android/blob/cb07541879b2484e5f4de90d91a80ecc12eb8588/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryScreen.kt#L41)

I'd really appreciate it if you could check those out and migrate them (it should be 15 minutes of code changes; you can look at the other usages of `WCModalBottomSheet` I've migrated already for reference) 🙏🏼 . Please let me know if you encounter any issues or I can help with any of those.

samiuelson commented:
> * [ScanToUpdateInventoryScreen](https://github.com/woocommerce/woocommerce-android/blob/cb07541879b2484e5f4de90d91a80ecc12eb8588/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryScreen.kt#L41)

Thanks for ping @JorgeMucientes. I added a [task](https://github.com/woocommerce/woocommerce-android/issues/13513) to Peacock's backlog for the scan to update inventory screen. We will take care of this one.

JFYI, scan to update inventory screen is accessible when clicking "scan barcode" icon in Products screen.

atorresveiga commented:
Thanks for the ping @JorgeMucientes. I added the task [here](https://github.com/woocommerce/woocommerce-android/issues/13396), where I can tackle with the rest of the other SDK 35 tasks

JorgeMucientes committed:
Merge branch 'trunk' into issue/13495-migrate-to-use-modebottomsheet-material3

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/themes/ThemePreviewScreen.kt

JorgeMucientes committed:
Minor padding adjustments

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/WCModalBottomSheet.kt:
```diff
@@ -0,0 +1,42 @@
+    content: @Composable ColumnScope.() -> Unit,
+) {
+    ModalBottomSheet(
+        containerColor = colorResource(R.color.color_surface_elevated),
```
Had to manually override this because the default Material 3 dark colors for bottom sheets are off: 

<img width="300"  src="https://github.com/user-attachments/assets/dbf12c71-22b2-485f-b3df-39e72b0a0ff2">

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/WCModalBottomSheet.kt:
```diff
@@ -0,0 +1,42 @@
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun WCModalBottomSheet(
```
WDYT about allowing consumers to override the shape? We can have a default one that will be used when no override is needed.

And also, the default shape of the Material 3 seems to have bigger corners than the rest of the app, WDYT about passing a specific shape there? Previously we used `RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp)`

hichamboushaba reviewed (approved):
Nice job @JorgeMucientes, code looks good and works well.

I left just one suggestion, and I'm pre-approving.

JorgeMucientes committed:
Enable shape ovveriding for WCModalBottomSheet

Additionally, set default value to 8.dp for consistency with the rest of the "corners" in the app
